reproducibility_project/src/engines/mcccs/conv_traj.py

```diff
@@ -0,0 +1,113 @@
+
+import freud
+import matplotlib.pyplot as plt
+import mdtraj as md
+import numpy as np
+import signac
+from scipy import stats
+
+
+def main():
+    """Read xyz files from each replica and combine them all into one big trajectory."""
+    data_path = "analysis_data"
+    # if os.path.exists(data_path):
+    #    shutil.rmtree(data_path)
+    # os.makedirs(data_path)
+    # delete the folder manually
+    if not os.path.isdir(data_path):
+
+        os.makedirs(data_path)
+
+    os.chdir(data_path)
+
+    project = signac.get_project()
+
+    for (molecule, ensemble, temperature, pressure), group in project.groupby(
+        (
+            "molecule",
+            "ensemble",
+            "temperature",
+            "pressure",
+        )
+    ):
+        print(molecule, ensemble, temperature, pressure)
+        if not os.path.isdir(
+            "{}_{}_{}K_{}kPa".format(molecule, ensemble, temperature, pressure)
+        ):
+
+            os.makedirs(
+                "{}_{}_{}K_{}kPa".format(
+                    molecule, ensemble, temperature, pressure
+                )
+            )
+        os.chdir(
+            "{}_{}_{}K_{}kPa".format(molecule, ensemble, temperature, pressure)
+        )
+        base_dir = os.getcwd()
+        if ensemble == "NPT" and molecule == "methaneUA":
+            if os.path.exists(os.path.join(os.getcwd(), "comb_traj.h5")):
+                continue
+            traj_list = []
+            for job in group:
+                # print(job)
+                os.chdir(job.ws)
+                traj_files = sorted(glob("box1movie1a*prod*pdb*"))
+                if len(traj_files) < 4:
+                    # print("warning, only {}  prod cycles complete for {} {} {} {} {}".format(len(traj_files),job, molecule, ensemble, temperature, pressure))
+                    print(
+                        "warning, only {}  prod cycles complete for {} {} {} {} {}. So not inlcuding its xyz files.".format(
+                            len(traj_files),
+                            job,
+                            molecule,
+                            ensemble,
+                            temperature,
+                            pressure,
+                        )
+                    )
+                    continue
+                prod_number = 0
+                for filename in traj_files:
+                    print(
+                        "The filename is {} and the prod number is {}. These two should match.".format(
+                            filename, prod_number
+                        )
+                    )
+                    traj = md.load(filename, top="init1.mol2")
+                    fort12_filename = "fort.12.prod{}".format(prod_number)
+                    fort12 = np.genfromtxt(fort12_filename, skip_header=1)
+                    traj = md.Trajectory(
+                        traj.xyz,
+                        traj.top,
+                        unitcell_lengths=fort12[9::10, 0:3],
```

Comment:
Do the unit cell lengths need to be converted from angstroms to nanometers?  I saw that you converted them to nm in the sanitized log files, but it this doesn't appear to be reading from the sanitized log files.

Reply:
From mdtraj docs 1.9.5
>There’s a lot of information in the trajectory object. The most obvious is the Cartesian coordinates. They’re stored as a numpy array under xyz. All of the distances in the Trajectory are stored in nanometers. The time unit is picoseconds. Angles are stored in degrees (not radians).

Just make sure you are doing the proper conversions/the loader is doing the proper conversion.

Reply:
Thanks. I had to convert them.